v2/federation/fetch.go

```diff
@@ -2,26 +2,71 @@ package federation
 
 import (
 	"context"
+	"net/http"
 
 	"github.com/spiffe/go-spiffe/v2/bundle/spiffebundle"
 	"github.com/spiffe/go-spiffe/v2/bundle/x509bundle"
 	"github.com/spiffe/go-spiffe/v2/spiffeid"
+	"github.com/spiffe/go-spiffe/v2/spiffetls/tlsconfig"
 	"github.com/zeebo/errs"
 )
 
 var federationErr = errs.Class("federation")
 
 // FetchOption is an option used when dialing the bundle endpoint.
-type FetchOption interface{}
+type FetchOption interface {
+	apply(*fetchOptions)
+}
+
+type fetchOptions struct {
+	transport http.RoundTripper
+}
+
+var defaultFetchOptions = fetchOptions{
+	transport: http.DefaultTransport,
+}
```

Comment:
Hmm, how would you feel about `transport` being a concrete `*http.Transport` instead of a round tripper? The default could then be a copy of the default http transport:

```
defaultTransport := *http.DefaultTransport.(*http.Transport)
transport = &defaultTransport
```

I think there is value in preserving the default transport behavior even when we replace the TLS config for SPIFFE auth.